url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/event/net/io/KyoroServerSocketImpl.java

```diff
@@ -0,0 +1,73 @@
+package net.hetimatan.util.event.net.io;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+
+import net.hetimatan.util.event.EventTask;
+
+public class KyoroServerSocketImpl extends KyoroServerSocket {
+
+	private ServerSocketChannel mServerChannel = null;
+	private boolean mIsBinded = false;
+
+	public KyoroServerSocketImpl() throws IOException {
+		 mServerChannel = ServerSocketChannel.open();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource that might leak to a class field, causing a resource leak that slows down or crashes your system. For example, when AbstractSelectableChannel.configureBlocking throws an exception, the *KyoroServerSocketImpl* object isn’t constructed so you can’t close this resource afterward.

**Fix**
Ensure that if an exception is thrown, the *mServerChannel* resource is closed before the constructor is exited.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).